### PR TITLE
Add support for JWK Sets

### PR DESCRIPTION
**Pull request by burrbd**
Nice package. Would you be open to adding support for [JWK Sets](https://tools.ietf.org/html/draft-ietf-jose-json-web-key-41#section-5)?

Thanks

### CODE REVIEW

File: jwk.go [hunk 1]
```diff
@@ -56,6 +56,67 @@ type JsonWebKey struct {
 	Algorithm string
 }
 
+// rawJsonWebKeySet represents a JWK Set with raw members.
+type rawJsonWebKeySet struct {
+	Keys []json.RawMessage `json:"keys"`
```
Comment (csstaub) [inline jwk.go]: If `JsonWebKey` implements `MarshalJSON` and `UnmarshalJSON`, would it not be possible to put `[]JsonWebKey` in here and have the `json` package parse it directly?

File: jwk_test.go [hunk 2]
```diff
@@ -414,3 +414,45 @@ func TestThumbprint(t *testing.T) {
...
+	jsonbar2, err := set2.MarshalJSON()
+	if err != nil {
+		t.Errorf("problem marshalling set", err)
+	}
+	if !bytes.Equal(jsonbar, jsonbar2) {
+		t.Error("roundtrip should not lose information")
+	}
+}
+
+func TestJWKSetKey(t *testing.T) {
```
Comment (csstaub) [inline jwk_test.go]: Maybe we should add a test to make sure this holds true: ```The member names within a JWK Set MUST be unique; JWK Set parsers
   MUST either reject JWK Sets with duplicate member names or use a JSON
   parser that returns only the lexically last duplicate member name, as
   specified in Section 15.12 (The JSON Object) of ECMAScript 5.1
   [ECMAScript].```

File: jwk.go [hunk 3]
```diff
@@ -56,6 +56,67 @@ type JsonWebKey struct {
...
+// UnmarshalJSON reads a set from its JSON representation.
+func (s *JsonWebKeySet) UnmarshalJSON(data []byte) error {
+	var rawSet rawJsonWebKeySet
+
+	if err := json.Unmarshal(data, &rawSet); err != nil {
+		return err
+	}
+
+	for _, raw := range rawSet.Keys {
+
```
Comment (csstaub) [inline jwk.go]: Nit: extra whitespace

**Comment by csstaub** [comment]
Hi! Yes, we can certainly add support for JWK sets. I added some comments to your code, but in general it looks pretty good. You'll have to sign our CLA before we can accept, see https://github.com/square/go-jose/blob/master/CONTRIBUTING.md

File: jwk.go (hunk 1 above)
Comment (burrbd) [inline jwk.go]: Wow cool. Wasn't aware of those interfaces. Makes code much simpler.

File: jwk_test.go (hunk 2 above)
Comment (burrbd) [inline jwk_test.go]: Okay. I've added a test for this.

**Comment by csstaub** [comment]
Looks good, thanks!

**Pull request merged**